src/buf_writer.c

```diff
@@ -104,24 +120,43 @@ buf_writer_flush(buf_writer_t *buf_writer) {
 	buf_writer->private_write_cb(buf_writer->private_cbopaque,
 	    buf_writer->buf);
 	buf_writer->buf_end = 0;
+	buf_writer_assert(buf_writer);
+}
+
+static void
+buf_writer_overflow(buf_writer_t *buf_writer) {
+	buf_writer_assert(buf_writer);
+	assert(!buf_writer->allow_auto_flush);
+	assert(buf_writer->buf_end == buf_writer->buf_size);
```

Comment:
This is only true because we have the unnecessary last iteration of memcpy, right? Let's remove if so.